src/core/job.c

```diff
@@ -418,7 +417,7 @@ JobType job_type_collapse(JobType t, Unit *u) {
 
         case JOB_TRY_RESTART:
                 s = unit_active_state(u);
-                if (!UNIT_IS_ACTIVE_OR_RELOADING(s))
+                if (!UNIT_IS_ACTIVE_OR_ACTIVATING(s))
```

Comment:
so, this will change that from now on JOB_TRY_RESTART will be collapsed into JOB_RESTART when UNIT_ACTIVATING state is in effect, as opposed to being suppressed to JOB_NOP.

I think a comment would be good here too, that clarifies that this is about not losing restart jobs enqueued while we are activating.